erica_app/erica/request_processing/requests_controller.py

```diff
@@ -88,29 +88,41 @@ def process(self):
         # the Elster specification (see `Jahresdokumentation_10_2020.xml`)
         fields = est_mapping.check_and_generate_entries(self.input_data.est_data.__dict__)
 
-        if self.input_data.est_data.new_admission:
+        if self.input_data.est_data.submission_without_tax_nr:
             empfaenger = self.input_data.est_data.bufa_nr
-            electronic_steuernummer = None
+            vorsatz = generate_vorsatz_without_tax_number(
+                self.input_data.meta_data.year,
+                self.input_data.est_data.person_a_idnr,
+                self.input_data.est_data.person_b_idnr,
+                self.input_data.est_data.person_a_first_name,
+                self.input_data.est_data.person_a_last_name,
+                self.input_data.est_data.person_a_street,
+                self.input_data.est_data.person_a_street_number,
+                self.input_data.est_data.person_a_plz,
+                self.input_data.est_data.person_a_town)
         else:
-            electronic_steuernummer = est_mapping.generate_electronic_steuernummer(self.input_data.est_data.steuernummer,
-                                                                               self.input_data.est_data.bundesland,
-                                                                               use_testmerker=self._is_testmerker_used())
+            electronic_steuernummer = est_mapping.generate_electronic_steuernummer(
+                self.input_data.est_data.steuernummer,
+                self.input_data.est_data.bundesland,
+                use_testmerker=self._is_testmerker_used())
+            vorsatz = generate_vorsatz_with_tax_number(
+                electronic_steuernummer,
+                self.input_data.meta_data.year,
+                self.input_data.est_data.person_a_idnr,
+                self.input_data.est_data.person_b_idnr,
+                self.input_data.est_data.person_a_first_name,
+                self.input_data.est_data.person_a_last_name,
+                self.input_data.est_data.person_a_street,
+                self.input_data.est_data.person_a_street_number,
+                self.input_data.est_data.person_a_plz,
+                self.input_data.est_data.person_a_town)
```

Comment:
We could make it more explicit what's shared and what isn't like this:

```python
        common_vorsatz_args = (
                self.input_data.meta_data.year,
                self.input_data.est_data.person_a_idnr,
                self.input_data.est_data.person_b_idnr,
                self.input_data.est_data.person_a_first_name,
                self.input_data.est_data.person_a_last_name,
                self.input_data.est_data.person_a_street,
                self.input_data.est_data.person_a_street_number,
                self.input_data.est_data.person_a_plz,
                self.input_data.est_data.person_a_town
        )
        if self.input_data.est_data.submission_without_tax_nr:
            empfaenger = self.input_data.est_data.bufa_nr
            vorsatz = generate_vorsatz_without_tax_number(*common_vorsatz_args)
        else:
            electronic_steuernummer = est_mapping.generate_electronic_steuernummer(
                self.input_data.est_data.steuernummer,
                self.input_data.est_data.bundesland,
                use_testmerker=self._is_testmerker_used())
            vorsatz = generate_vorsatz_with_tax_number(electronic_steuernummer, *common_vorsatz_args)
```

WDYT? Makes it pretty clear that the only difference is that one gets the electro-tax-nr, the other doesn't.

Reply:
Great idea!